controllers/storagecluster/reconcile.go

```diff
@@ -389,6 +389,19 @@ func (r *StorageClusterReconciler) reconcilePhases(
 			&ocsSnapshotClass{},
 			&ocsNoobaaSystem{},
 		}
+
+		if isExternalOCSProvider(instance) {
+			grpcConnection, externalClusterClient, err := newExternalClusterClient(instance)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			r.externalClusterClient = externalClusterClient
+			defer func() {
+				grpcConnection.Close()
```

Comment:
This should be moved into newExternalClusterClient function

Reply:
It can not be moved as defer will be called exactly after newExternalClusterClient is done and we will lost the connection

Reply:
But don't we need the grpcConnection later?